controllers/vspheremachine_controller.go

```diff
@@ -151,35 +157,98 @@ func (r machineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr er
 			vsphereMachine.Name)
 	}
 
-	// Create the cluster context for this request.
-	clusterContext := &context.ClusterContext{
-		ControllerContext: r.ControllerContext,
-		Cluster:           cluster,
-		VSphereCluster:    vsphereCluster,
-		Logger:            r.Logger.WithName(req.Namespace).WithName(req.Name),
-		PatchHelper:       patchHelper,
-	}
-
 	// Create the machine context for this request.
 	machineContext := &context.MachineContext{
-		ClusterContext: clusterContext,
+		ClusterContext: &context.ClusterContext{
+			ControllerContext: r.ControllerContext,
+			Cluster:           cluster,
+			VSphereCluster:    vsphereCluster,
+		},
 		Machine:        machine,
 		VSphereMachine: vsphereMachine,
 		Session:        authSession,
 		Logger:         r.Logger.WithName(req.Namespace).WithName(req.Name),
 		PatchHelper:    patchHelper,
 	}
 
+	// Print the task-ref upon entry and upon exit.
+	machineContext.Logger.V(4).Info(
+		"VSphereMachine.Status.TaskRef OnEntry",
+		"task-ref", machineContext.VSphereMachine.Status.TaskRef)
+	defer func() {
+		machineContext.Logger.V(4).Info(
+			"VSphereMachine.Status.TaskRef OnExit",
+			"task-ref", machineContext.VSphereMachine.Status.TaskRef)
+	}()
+
 	// Always issue a patch when exiting this function so changes to the
 	// resource are patched back to the API server.
 	defer func() {
+		// Patch the VSphereMachine resource.
 		if err := machineContext.Patch(); err != nil {
 			if reterr == nil {
 				reterr = err
-			} else {
-				machineContext.Logger.Error(err, "patch failed", "machine", machineContext.String())
 			}
+			machineContext.Logger.Error(err, "patch failed", "machine", machineContext.String())
 		}
+
+		// localObj references the VSphereMachine resource fetched at the
+		// beginning of this reconcile request.
+		localObj := machineContext.VSphereMachine.DeepCopy()
```

Comment:
I had you put in the DeepCopy() here when we were debugging, but it turns out it's not necessary. And I would actually change this to store `machineContext.VSphereMachine.ResourceVersion` - you shouldn't need the full object.

Reply:
Ah, good point. I didn't get rid of the deep copy after I removed the logging. Storing just the RV will be simpler for sure.